functions/src/custom/totals.ts

```diff
@@ -0,0 +1,76 @@
+import * as functions from "firebase-functions";
+import * as Sentry from "@sentry/node";
+import logger from "../services/logging";
+import { firestore } from "../admin/admin";
+import { environment } from "../environment";
+
+const event_collection = environment.FIRESTORE_EVENT_COLLECTION as string;
+
+export const computeCollectionTotals = firestore.listCollections().then(async (collections) => {
+  const total: Record<string, number> = {};
+  const collectionTotalRef = firestore.collection("total").doc("collection_total");
+
+  for (const collection of collections) {
+    const collectionId = collection.id;
+    collection.listDocuments().then((documents) => {
+      total[collectionId] = documents.length;
+    });
+    try {
+      await collectionTotalRef.update(total);
+      logger.log({
+        total,
+        message: `Total number of documents in collection ${collectionId} is ${total[collectionId]}`,
+      });
+    } catch (error) {
+      Sentry.captureException(error);
+      logger.log(`Failed to fetch total number of documents in collection ${collectionId}`);
+    }
+  }
+});
+
+export const fetchParticipantCount = firestore
+  .collection(event_collection)
+  .listDocuments()
+  .then(async (documents) => {
+    const attendanceCountForDocument: Record<string, number> = {};
+    const eventTotalRef = firestore.collection("total").doc("event");
+
+    for (const document of documents) {
+      const documentRef = await document.get();
+      const attendanceCount = documentRef.get("attendance").length;
+      attendanceCountForDocument[documentRef.id] = attendanceCount;
+
+      try {
+        await eventTotalRef.update(attendanceCountForDocument);
+        logger.log({
+          attendanceCountForDocument,
+          message: `Total attendance for event ${documentRef.id} is ${attendanceCount}`,
+        });
+      } catch (error) {
+        Sentry.captureException(error);
+        logger.log(`Failed to fetch total attendance for event ${documentRef.id}`);
+      }
+    }
+  });
+
+export const newAttendanceTrigger = functions.firestore
```

Comment:
Firestore triggers need to be registered in `/functions/src/index.ts`for the firebase to actually use them. See examples [here](https://github.com/acmutd/portal-backend/blob/8e7783bd8612e00372cf5661d5d6298ea8f44e3e/functions/src/index.ts#L21).

Reply:
Just registered the scripts and triggers in the latest commit.